rpc/backend/evm_backend_test.go

```diff
@@ -1,30 +1,380 @@
+			},
+			ethtypes.Bloom{},
+			false,
+		},
+		{
+			"fail - nonblock bloom attribute key",
+			&tmrpctypes.ResultBlockResults{
+				EndBlockEvents: []types.Event{
+					{
+						Type: evmtypes.EventTypeBlockBloom,
+						Attributes: []types.EventAttribute{
+							{Key: []byte(evmtypes.AttributeKeyEthereumTxHash)},
+						},
+					},
+				},
+			},
+			ethtypes.Bloom{},
+			false,
+		},
+		{
+			"pass - nonblock bloom attribute key",
+			&tmrpctypes.ResultBlockResults{
+				EndBlockEvents: []types.Event{
+					{
+						Type: evmtypes.EventTypeBlockBloom,
+						Attributes: []types.EventAttribute{
+							{Key: []byte(bAttributeKeyEthereumBloom)},
+						},
+					},
+				},
+			},
+			ethtypes.Bloom{},
+			true,
+		},
+	}
+	for _, tc := range testCases {
+		blockBloom, err := suite.backend.BlockBloom(tc.blockRes)
+
+		if tc.expPass {
+			suite.Require().Nil(err)
+			suite.Require().Equal(tc.expBlockBloom, blockBloom)
+		} else {
+			suite.Require().NotNil(err)
+		}
+	}
+}
+
+func (suite *BackendTestSuite) TestBaseFee() {
+	baseFee := sdk.NewInt(1)
+
+	testCases := []struct {
+		mame                string
+		blockRes            *tmrpctypes.ResultBlockResults
+		registerMockQueries func()
+		expBaseFee          *big.Int
+		expPass             bool
+	}{
+		{
+			"fail - grpc BaseFee error",
+			&tmrpctypes.ResultBlockResults{Height: 1},
+			func() {
+				queryClient := suite.backend.queryClient.QueryClient.(*mocks.QueryClient)
+				RegisterBaseFeeQueriesError(queryClient)
+			},
+			nil,
+			false,
+		},
+		{
+			"fail - grpc BaseFee error - with non feeemarket block event",
+			&tmrpctypes.ResultBlockResults{
+				Height: 1,
+				BeginBlockEvents: []types.Event{
+					{
+						Type: evmtypes.EventTypeBlockBloom,
+					},
+				},
+			},
+			func() {
+				queryClient := suite.backend.queryClient.QueryClient.(*mocks.QueryClient)
+				RegisterBaseFeeQueriesError(queryClient)
+			},
+			nil,
+			false,
+		},
+		{
+			"fail - grpc BaseFee error - with feeemarket block event",
+			&tmrpctypes.ResultBlockResults{
+				Height: 1,
+				BeginBlockEvents: []types.Event{
+					{
+						Type: feemarkettypes.EventTypeFeeMarket,
+					},
+				},
+			},
+			func() {
+				queryClient := suite.backend.queryClient.QueryClient.(*mocks.QueryClient)
+				RegisterBaseFeeQueriesError(queryClient)
+			},
+			nil,
+			false,
+		},
+		{
+			"fail - grpc BaseFee error - with feeemarket block event with wrong attribute value",
+			&tmrpctypes.ResultBlockResults{
+				Height: 1,
+				BeginBlockEvents: []types.Event{
+					{
+						Type: feemarkettypes.EventTypeFeeMarket,
+						Attributes: []types.EventAttribute{
+							{Value: []byte{0x1}},
+						},
+					},
+				},
+			},
+			func() {
+				queryClient := suite.backend.queryClient.QueryClient.(*mocks.QueryClient)
+				RegisterBaseFeeQueriesError(queryClient)
+			},
+			nil,
+			false,
+		},
+		{
+			"fail - grpc BaseFee error - with feeemarket block event with baseFee attribute value",
```

Comment:
```suggestion
			"fail - grpc baseFee error - with feemarket block event with baseFee attribute value",
```

Other occurrences too